packages/test/service-load-test/src/loadTestDataStore.ts

```diff
@@ -18,113 +23,215 @@ export interface IRunConfig {
+        root.set(taskManagerKey, TaskManager.create(runtime).handle);
     }
 
-    public async pause(timeMs: number) {
-        const startTimeMs = Date.now();
-        this.state = "paused";
-        await wait(timeMs);
-        this.state = "running";
-        return Date.now() - startTimeMs;
+    private static async waitForCatchup(runtime: IFluidDataStoreRuntime) {
+        if(runtime.deltaManager.active) {
+            return;
+        }
+        const lastKnownSeq = runtime.deltaManager.lastKnownSeqNumber;
+        while(runtime.deltaManager.lastSequenceNumber < lastKnownSeq) {
+            await new Promise((resolve,reject)=>{
+                if(runtime.disposed) {
+                    reject(new Error("disposed"));
+                    return;
+                }
+                runtime.deltaManager.once("op", resolve);
+            });
+        }
     }
 
-    private printStatus(config: IRunConfig, startTimeMs: number, runningStartTimeMs: number) {
-        const now = Date.now();
-        const totalMin = (now - startTimeMs) / 60000;
-        const runningMin = (now - runningStartTimeMs) / 60000;
-        const opRate = Math.floor(this.opCount / totalMin);
-        const sendRate = Math.floor(this.sentCount / runningMin);
-        console.log(
-            `${config.runId.toString().padStart(3)}>` +
-            ` seen: ${this.opCount.toString().padStart(8)} (${opRate.toString().padStart(4)}/min),` +
-            ` sent: ${this.sentCount.toString().padStart(8)} (${sendRate.toString().padStart(2)}/min),` +
-            ` run time: ${runningMin.toFixed(2).toString().padStart(5)} min`,
-            ` total time: ${totalMin.toFixed(2).toString().padStart(5)} min`,
+    public static async createRunnerInstance(
+        config: IRunConfig,
+        reset: boolean,
+        root: ISharedDirectory,
+        runtime: IFluidDataStoreRuntime,
+    ) {
+        if(!root.hasSubDirectory(config.runId.toString())) {
+            root.createSubDirectory(config.runId.toString());
+        }
+        const runDir = root.getSubDirectory(config.runId.toString());
+        if(runDir === undefined) {
+            throw new Error(`runDir for runId ${config.runId} not available`);
+        }
+
+        if(!runDir.has(counterKey)) {
+            await LoadTestDataStoreModel.waitForCatchup(runtime);
+            if(!runDir.has(counterKey)) {
+                runDir.set(counterKey, SharedCounter.create(runtime).handle);
+                runDir.set(startTimeKey,Date.now());
+            }
+        }
+        const counter = await runDir.get<IFluidHandle<ISharedCounter>>(counterKey)?.get();
+        const taskmanager = await root.wait<IFluidHandle<ITaskManager>>(taskManagerKey).then(async (h)=>h.get());
+
+        if(counter === undefined) {
+            throw new Error("counter not available");
+        }
+        if(taskmanager === undefined) {
+            throw new Error("taskmanger not available");
+        }
+
+        if(reset) {
```

Comment:
Do we not need to do any cleanup of an old runner in this case?

Reply:
that what this does. i'm not sure what else you are referring too

Reply:
Ok, I see now.  The model class doesn't maintain ownership of any state, it just keeps references to the DDS's held by the DataObject. So all its persistent state is encapsulated by the DDS's, and swapping instances of it has no fallout.  I was concerned that we were returning a new instance of the object without doing any teardown of the old one.

And at any point the DataObject could just do `dataModel = await LoadTestDataStoreModel.createRunnerInstance(
            config, reset, this.root, this.runtime);` and it would be good to go again.

Seems like `reset` param isn't needed, and should always be considered true?